platforms/documentation/docs/src/snippets/dependencyManagement/customizingResolution-secondaryVariants/groovy/build.gradle

```
@@ -0,0 +1,52 @@
+// tag::lib[]
+plugins {
+    id 'java-library'
+}
+
+group = 'com.gradle'
+version = '1.2.3'
+// end::lib[]
+
+// tag::add[]
+// Create a new secondary variant on the existing 'apiElements' configuration.
+// This variant will be available during artifact selection (not graph resolution).
+configurations.apiElements.outgoing.variants.create("customApiElement") {
+    attributes {
+        attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage, "custom-variant"))
+    }
+}
+// end::add[]
+
+// tag::custom[]
+// Define custom attributes for variant identification
+def fooAttribute = Attribute.of("FooAttribute", String)
+def fooVariantAttribute = Attribute.of("FooVariantAttribute", String)
```

```suggestion
def fooAttribute = Attribute.of("com.example.foo", String)
def fooVariantAttribute = Attribute.of("com.example.fooVariant", String)
```

We should be promoting attribute names with a java-like namespace pattern, as this prevents conflicts between ecosystems, and is the standard already.

---

Fix in Kotlin and Groovy.